src/app/shared/popup/popup-api.service.ts

```diff
@@ -0,0 +1,20 @@
+import {Injectable, TemplateRef} from '@angular/core';
+import {Subject} from 'rxjs';
+
+@Injectable({
+    providedIn: 'root',
+})
+export class PopupService {
+    readonly popupSubject$ = new Subject<{
+        template: TemplateRef<{$implicit: string}>;
+        context: string;
+    } | null>();
```

Comment:
Лучше не давать полного доступа к сабжектам, ведт тогда пропадает инкапсулирование хранилища и сложно следить за корректностью его изменений. Стоит ограничивать доступ к сабжектам только для чтения, а позволять изменять только через методы сервиса. Этого можно добиться путем приватизации свойства `popupSubject$` и отдачу во вне потока по нему
```suggestion
    private readonly popupSubject$ = new Subject<{
        template: TemplateRef<{$implicit: string}>;
        context: string;
    } | null>();
    
    readonly popup$ = this.popupSubject$.asObservable();
```